vllm/v1/core/kv_cache_manager.py

```diff
@@ -203,33 +205,28 @@ def allocate_slots(
             self.free_block_queue.num_free_blocks -
             num_evictable_computed_blocks)
 
-        num_computed_tokens = len(computed_blocks) * self.block_size
-
-        # When caching is enabled, get the new token IDs and the parent block
-        # ID to generate cache keys.
-        new_token_ids = None
-        parent_block = None
-        if self.enable_caching:
-            # Touch the computed blocks to make sure they won't be evicted.
-            self._touch(computed_blocks)
-
-            # Get the token IDs for the blocks being allocated for hashing.
-            new_token_ids = request.all_token_ids[
-                num_computed_tokens:num_computed_tokens + num_tokens]
-            if not new_token_ids:
-                raise RuntimeError(
-                    "Failed to infer the token IDs for allocation. "
-                    f"#all_tokens={len(request.all_token_ids)} < "
-                    f"#computed_tokens={num_computed_tokens}")
+        # Concatenate the computed block IDs and the new block IDs.
+        new_blocks = self._get_new_blocks(num_new_blocks)
+        self.req_to_blocks[request.request_id] = computed_blocks + new_blocks
 
-            # Get the parent block ID to construct the block chain.
-            parent_block = computed_blocks[-1] if computed_blocks else None
+        if not self.enable_caching:
+            return new_blocks
 
-        new_blocks = self._get_new_blocks(num_new_blocks, new_token_ids,
-                                          parent_block)
+        self._touch(computed_blocks)
```

Comment:
We should touch before getting new blocks; otherwise the computed blocks may be evicted before being touched.

Reply:
oh right - good catch.

Reply:
added a test.